### PR TITLE
Subnet delegation lat

### PR DESCRIPTION
**Pull request by wdehrich**
## Description
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/147

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 1]
```diff
@@ -451,6 +451,11 @@
     <Compile Include="Generated\RouteTable\Route\RemoveAzureRmRouteConfigCommand.cs" />
     <Compile Include="Generated\RouteTable\Route\SetAzureRmRouteConfigCommand.cs" />
     <Compile Include="Generated\RouteTable\SetAzureRMRouteTableCommand.cs" />
+    <Compile Include="SubnetDelegation\AzureSubnetDelegationBaseCmdlet.cs" />
+    <Compile Include="SubnetDelegation\GetAzureServiceEndpointPolicyCommand.cs" />
```
Comment (chandrasekarsrinivasan) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: remove this

File: src/ResourceManager/Network/Commands.Network/Microsoft.Azure.Commands.Network.format.ps1xml [hunk 2]
```diff
@@ -1139,6 +1139,14 @@
               <ListItem>
                 <Label>EnableIPForwarding</Label>
                 <PropertyName>EnableIPForwarding</PropertyName>
+              </ListItem>
+              <ListItem>
+                <Label>HostedWorkloads</Label>
```
Comment (chandrasekarsrinivasan) [inline src/ResourceManager/Network/Commands.Network/Microsoft.Azure.Commands.Network.format.ps1xml]: HostedWorkLoad

File: src/ResourceManager/Network/Commands.Network/Models/PSNetworkInterface.cs [hunk 3]
```diff
@@ -39,6 +39,12 @@ public class PSNetworkInterface : PSTopLevelResource, INetworkInterfaceReference
         [Ps1Xml(Target = ViewControl.Table)]
         public bool? EnableIPForwarding { get; set; }
 
+        [Ps1Xml(Target = ViewControl.Table)]
+        public List<string> HostedWorkloads { get; set; }
```
Comment (chandrasekarsrinivasan) [inline src/ResourceManager/Network/Commands.Network/Models/PSNetworkInterface.cs]: Singular

**Comment by praries880** [comment]
the following failing tests need to be fixed : 

Desktop : https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/7262/artifact/src/Publish/TestResults/FailingTests/


Netcore : https://travis-ci.org/Azure/azure-powershell/builds/425543468#L2755

The iossue in all of them sees to be the inability of CI to load the network SDK dll. Is the library signed?

**Comment by praries880** [comment]
Tests failing because of API version change : https://travis-ci.org/Azure/azure-powershell/builds/425913730#L1591

You'll need to re-record the tests.

Contact @chandrasekarsrinivasan for the current effort to update the network test recordings.

**Comment by praries880** [comment]
@wdehrich  Changes look good to me, The test recordings need to be uipdated after the version bump in NRP SDK.

We can merge this once the tests are fixed.

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="3.4.0-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.3.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.5.0-preview" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj]: .NET SDK is merged and (with a bit of magic) finally published https://www.nuget.org/packages/Microsoft.Azure.Management.Network/19.4.0-preview 

So now all contributors must change SDK version from local version (localFeed folder) to 19.4.0 and remove nuget packages from localFeed folder.

File: src/ResourceManager/Network/Commands.Network/Models/PSNetworkInterface.cs [hunk 5]
```diff
@@ -39,6 +39,9 @@ public class PSNetworkInterface : PSTopLevelResource, INetworkInterfaceReference
         [Ps1Xml(Target = ViewControl.Table)]
         public bool? EnableIPForwarding { get; set; }
 
+        [Ps1Xml(Target = ViewControl.Table)]
```
[the remark this answers is not in the document]
Comment (wdehrich) [reply src/ResourceManager/Network/Commands.Network/Models/PSNetworkInterface.cs]: It was deliberately removed

**Comment by praries880** [comment]
@wdehrich  you need to remove these files form the localfeed : 
tools/LocalFeed/Microsoft.Azure.Management.Network.19.5.0-preview.nupkg 
and
tools/LocalFeed/Microsoft.Azure.Management.Network.19.5.0-preview.symbols.nupkg

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj [hunk 6]
```diff
@@ -25,7 +25,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="3.4.0-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.3.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.4.0-preview" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj]: did you pull from "Azure:network-september-release" and merge the altest changes in the ranch with yours?
I am surprised that you are seeing the version changes if you already merged

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 7]
```diff
@@ -20,6 +20,7 @@
 
 ## Current Release
 * Updated cmdlet New-AzureRmVirtualNetworkGatewayConnection with support for switch ExpressRouteGatewayBypass
+* Added new HostedWorkloads property for PSNetworkInterface
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ChangeLog.md]: @wdehrich  It does not seem that the altest changes in Azure:network-september-release have been ,erged here. Can you merge the upstream branch and update the PR?

**Comment by praries880** [comment]
closing this in favor of https://github.com/Azure/azure-powershell/pull/7223

**Pull request closed without merging**